File: B_PL_Codes/15-Ponteiros_basico/pont04.c
/*
Lógica de programação
Prof. Jonatha Costa
Exercício resolvido com ponteiros.
Escreva um código em C que preencha um vetor com 10 números e retorne quais são os números primos deste veto
*/

#include <stdio.h>

int main() {
    int v[10], i, j, primo, *p;

    printf("Digite 10 números:\n");
    for (p = v; p < v + 10; p++) {
        scanf("%d", p);
    }

    printf("Números primos:\n");
    for (p = v; p < v + 10; p++) {
        if (*p < 2) continue;
        primo = 1;
        for (j = 2; j <= *p / 2; j++) {
            if (*p % j == 0) {
                primo = 0;
                break;
            }
        }
        if (primo) printf("%d ", *p);
    }

    printf("\n");
    return 0;
}

File: B_PL_Codes/15-Ponteiros_basico/InJava/pont04.java
/*
Lógica de programação
Prof. Jonatha Costa
Exercício resolvido em java
Escreva um código em C que preencha um vetor com 10 números e retorne quais são os números primos deste veto
*/

import java.util.Scanner;

public class pont04 {

    public static void main(String[] args) {
        Scanner dados = new Scanner(System.in);

        int[] v = new int[10];
        boolean primo;

        System.out.printf("Digite 10 números:\n");
        for (int i = 0; i < 10; i++) {
            v[i] = dados.nextInt();
        }

        System.out.printf("Números primos:\n");
        for (int i = 0; i < 10; i++) {
            if (v[i] < 2)
                continue;
            primo = true;
            for (int j = 2; j <= v[i] / 2; j++) {
                if (v[i] % j == 0) {
                    primo = false;
                    break;
                }
            }
            if (primo)
                System.out.printf("%d ", v[i]);
        }

        System.out.printf("\n");
        dados.close();
    }
}
